ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -88,12 +89,13 @@
   private final ActivenessRegistrar activenessRegistrar;
   private final RequestValidator validator;
   private final RequestHandler handler;
+  private final Supplier<ServiceContext> serviceContextFactory;
 
 
   public KsqlResource(
       final KsqlConfig ksqlConfig,
       final KsqlEngine ksqlEngine,
-      final ServiceContext serviceContext,
+      final Supplier<ServiceContext> serviceContextFactory,
```

Comment:
Will this be a Supplier longer term? Seems like we want to pass in some info about the principal associated with the request.

Reply:
I initially did not add the supplier. There was a new DefaultServiceContext created on every request. However, I did not know how to verify it on the unit tests. So, I ended up using this supplier for that.

The plan for the next patch is to use a Function instead, where I can pass user information, like `serviceContextFactory.apply(principal)`.